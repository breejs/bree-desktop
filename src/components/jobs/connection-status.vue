<script setup>
import JobStatus from './job-status.vue';

const props = defineProps({
  status: {
    validator(value) {
      return ['running', 'waiting', 'stopped', 'error'].includes(value);
    },
    type: String,
    required: true
  }
});
</script>

<template lang="pug">
JobStatus(:kind='"connection"', :status='props.status')
</template>
