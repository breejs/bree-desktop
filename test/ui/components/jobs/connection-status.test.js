import { storyTests } from '../../_story-tests';

import * as stories from '@/components/jobs/connection-status.stories';

storyTests(stories);
