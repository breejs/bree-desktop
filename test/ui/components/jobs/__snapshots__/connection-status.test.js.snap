// Vitest Snapshot v1

exports[`story snapshots > Active 1`] = `"<i class=\\"bi fs-3 text-active bi-card-list\\" data-v-3b64800a=\\"\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Active\\" aria-label=\\"Active\\"></i>"`;

exports[`story snapshots > Delayed 1`] = `"<i class=\\"bi fs-3 text-delayed bi-card-list\\" data-v-3b64800a=\\"\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Delayed\\" aria-label=\\"Delayed\\"></i>"`;

exports[`story snapshots > Done 1`] = `"<i class=\\"bi fs-3 text-done bi-card-list\\" data-v-3b64800a=\\"\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Done\\" aria-label=\\"Done\\"></i>"`;

exports[`story snapshots > Error 1`] = `"<i class=\\"bi fs-3 text-error bi-card-list\\" data-v-3b64800a=\\"\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Error\\" aria-label=\\"Error\\"></i>"`;

exports[`story snapshots > Waiting 1`] = `"<i class=\\"bi fs-3 text-waiting bi-card-list\\" data-v-3b64800a=\\"\\" data-bs-toggle=\\"tooltip\\" data-bs-placement=\\"bottom\\" title=\\"\\" data-bs-original-title=\\"Waiting\\" aria-label=\\"Waiting\\"></i>"`;
